src/client/Microsoft.Identity.Client/Internal/Requests/OnBehalfOfRequest.cs

```diff
@@ -44,11 +44,15 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             MsalAccessTokenCacheItem msalAccessTokenItem = await CacheManager.FindAccessTokenAsync().ConfigureAwait(false);
             if (msalAccessTokenItem != null)
             {
+                var msalIdTokenItem = await CacheManager.GetIdTokenCacheItemAsync(msalAccessTokenItem.GetIdTokenItemKey()).ConfigureAwait(false);
```

Comment:
We already do this on SilentRequest: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/d36d5867d1af9f969b6aed805d746c650f9106f0/src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentClientAuthStrategy.cs#L124